sound/soc/sof/core.c

```diff
@@ -246,8 +261,6 @@ static int sof_probe_continue(struct snd_sof_dev *sdev)
 
 	return 0;
 
-fw_trace_err:
-	snd_sof_free_trace(sdev);
```

Comment:
I think the trace does not belong in the clients but in the core.
Not sure why it's removed here and not in snd_sof_device_remove() below.

Reply:
@plbossart its removed here because previously machine register could fail but there's not failure after its moved to the audio client. But it still needs to be freed when the SOF device is removed.